pydantic/_internal/_generate_schema.py

```diff
@@ -813,9 +813,12 @@ def _generate_schema_from_property(self, obj: Any, source: Any) -> core_schema.C
                 source, CallbackGetCoreSchemaHandler(self._generate_schema_inner, self, ref_mode=ref_mode)
             )
         elif (
-            (existing_schema := getattr(obj, '__pydantic_core_schema__', None)) is not None
+            hasattr(obj, '__dict__')
```

Comment:
We could also check for `__slots__` but there's no such type with a `__pydantic_core_schema__` property as of today, so I'm keeping things simple here by only checking for `__dict__`